src/common/database.c

```diff
@@ -2728,17 +2745,25 @@ int _get_pragma_val(const struct dt_database_t *db, const char* pragma)
 
 void dt_database_maybe_maintenance(const struct dt_database_t *db, const gboolean has_gui, const gboolean closing_time)
 {
-  gboolean check_for_maintenance = FALSE;
-
   char *config = dt_conf_get_string("database/maintenance_check");
 
+  if(!g_strcmp0(config, "never")){
+    // early bail out on "never"
+    fprintf(stderr, "[db maintenance] please consider enabling database maintenance.\n");
+    return;
+  }
+
+  gboolean check_for_maintenance = FALSE;
+  gboolean force_maintenance = g_str_has_suffix (config, "(don't ask)");
+
   if(config)
   {
-    if((!g_strcmp0(config, "on both"))
-        || (closing_time && !g_strcmp0(config, "on close"))
-        || (!closing_time && !g_strcmp0(config, "on startup")))
+    if((g_strstr_len(config, -1, "on both")) // should cover "(don't ask) suffix
```

Comment:
g_strstr_len() with -1 is equivalent to simple g_strstr() so better use it.

Reply:
There's no `g_strstr`:
`error: implicit declaration of function ‘g_strstr’; did you mean ‘g_strrstr’? [-Werror=implicit-function-declaration]`

That's what set me off initially... but `strstr` is there, so I can use it.